instantly/parse.go

```diff
@@ -0,0 +1,72 @@
+package instantly
+
+import (
+	"strconv"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+	"github.com/spyzhov/ajson"
+)
+
+func makeGetTotalSize(dataLocation string) common.ListSizeFunc {
+	return func(node *ajson.Node) (int64, error) {
+		return jsonquery.New(node).ArraySize(dataLocation)
+	}
+}
+
+func makeGetRecords(dataLocation string) common.RecordsFunc {
+	return func(node *ajson.Node) ([]map[string]any, error) {
+		arr, err := jsonquery.New(node).Array(dataLocation, false)
+		if err != nil {
+			return nil, err
+		}
+
+		return jsonquery.Convertor.ArrayToMap(arr)
+	}
+}
+
+func makeNextRecordsURL(dataLocation string, reqLink *urlbuilder.URL) common.NextPageFunc {
+	return func(node *ajson.Node) (string, error) {
+		size, err := makeGetTotalSize(dataLocation)(node)
+		if err != nil {
+			return "", err
+		}
+
+		if size < DefaultPageSize {
+			// There is no need to go further, definitely we are at the end.
+			return "", nil
+		}
+
+		previousStart := 0
+
+		skipQP, ok := reqLink.GetSingleQueryParam("skip")
+		if ok {
+			// Try to use previous "skip" parameter to determine the next skip.
+			skipNum, err := strconv.Atoi(skipQP)
+			if err == nil {
+				previousStart = skipNum
+			}
+		}
+
+		nextStart := previousStart + DefaultPageSize
+
+		reqLink.WithQueryParam("limit", strconv.Itoa(DefaultPageSize))
+		reqLink.WithQueryParam("skip", strconv.Itoa(nextStart))
+
+		return reqLink.String(), nil
+	}
+}
+
+func getMarshaledData(records []map[string]interface{}, fields []string) ([]common.ReadResultRow, error) {
```

Comment:
same as other functionss, define in `common` and remove duplicate definitions.

Reply:
Please, review: https://github.com/amp-labs/connectors/pull/971